core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java

```diff
@@ -60,13 +60,21 @@ private void distributeFeeToValidator(Address lendingPoolCoreAddr,BigInteger amo
 
         Map<String, BigInteger> ommValidators = call(Map.class, Contracts.STAKING,
                 "getActualUserDelegationPercentage", lendingPoolCoreAddr);
+        BigInteger amountToDistribute = amount;
+        BigInteger denominator = BigInteger.valueOf(100).multiply(ICX);
+        Address validatorAddr = lendingPoolCoreAddr;
         for (String validator : ommValidators.keySet()) {
-            Address validatorAddr = Address.fromString(validator);
-            BigInteger feePortion = (ommValidators.get(validator).multiply(amount)).divide(ICX).divide(BigInteger.valueOf(100));
+            validatorAddr = Address.fromString(validator);
+            BigInteger feePortion = (ommValidators.get(validator).multiply(amount)).divide(denominator);
+            amountToDistribute = amountToDistribute.subtract(feePortion);
```

Comment:
This implementation assumes to distribute the remaining amount to last validator. But it's better to use the progressive distribution  You would need to keep on deducting the total amount to distribute and the total(denominator).

Reply:
Got it. Applied the changes and pushed. Please review it again